admin/src/scenes/dashboard/inscription/ExportAll.js

```diff
@@ -1,75 +1,116 @@
 import React, { useState } from "react";
 import * as FileSaver from "file-saver";
 import * as XLSX from "xlsx";
+import { toastr } from "react-redux-toastr";
 
-import { departmentLookUp, department2region } from "../../../utils";
+import { departmentLookUp, department2region, departmentToAcademy, translate, translateModelFields } from "../../../utils";
 import LoadingButton from "../../../components/buttons/LoadingButton";
 import api from "../../../services/api";
+import ModalConfirm from "../../../components/modals/ModalConfirm";
 
 // TODO (fix): Export not only for 2021.
-export default function ExportAll() {
+export default function ExportAll({ filter }) {
   const [loading, setLoading] = useState(false);
+  const [modal, setModal] = useState({ isOpen: false, onConfirm: null });
+  const [loadingText, setLoadingText] = useState();
+
+  const onClick = () => {
+    setModal({
+      isOpen: true,
+      onConfirm: run,
+      title: "Téléchargement",
+      message:
+        "En téléchargeant ces informations, vous vous engagez à les supprimer après consultation en application des dispositions légales sur la protection des données personnelles (RGPD, CNIL)",
+    });
+  };
+
   async function run() {
-    const { data: inscriptionGoal } = await api.get("/inscription-goal/2021");
-    setLoading(true);
-    const lines = [];
+    if (!filter?.cohort?.length) return toastr.error("Merci de selectionner au moins une cohorte ");
+    if (!filter?.status?.length) return toastr.error("Merci de selectionner au moins un statut");
 
-    const dates = [];
-    const d = new Date(Date.UTC(2021, 1, 7, 3, 0, 0));
-    const limit = new Date();
-    limit.setDate(limit.getDate() - 7);
+    // starting the process...
+    setLoading(true);
 
-    while (d < limit) {
-      dates.push(new Date(d));
-      d.setDate(d.getDate() + 7);
-    }
-    dates.push(new Date(d));
-    d.setDate(d.getDate() + 7);
-    dates.push(new Date(d));
+    // get all the inscriptions goals for the selected cohorts
+    const inscriptionGoalsRequest = filter?.cohort?.map((cohort) => api.get(`/inscription-goal/${encodeURI(cohort)}`));
+    let inscriptionGoals = await Promise.all([...inscriptionGoalsRequest]);
+    inscriptionGoals = inscriptionGoals?.reduce((previous, current) => {
+      if (current.ok && current.data?.length) {
+        previous.push(...current.data);
+      }
+      return previous;
+    }, []);
+    const lines = [];
 
-    lines.push(["Région", "Numéro dtp", "Département", "Cible", ...dates.map((e) => e.toISOString().slice(0, 10))]);
+    lines.push(["Académie", "Région", "Numéro dpt", "Département", "Objectif", ...filter?.status?.map((e) => translate(e))]);
```

Comment:
Pour éviter un bug on peut faire genre ça, sinon le truc peut ne pas être itérable (non testé). C'était signalé parfois par eslint je crois
```suggestion
    lines.push(["Académie", "Région", "Numéro dpt", "Département", "Objectif", ...[filter?.status || []].map((e) => translate(e))]);
```